src/main/groovy/org/gradle/api/plugins/clover/AggregateReportsWithHistoryTask.groovy

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2011 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.plugins.clover
+
+import java.io.File;
+import java.util.List;
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.tasks.InputDirectory;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.OutputDirectory;
+import org.gradle.api.tasks.TaskAction
+
+/**
+ * Task for aggregrating Clover code coverage reports including historical reports.
+ *
+ * @author Christian Abegg
+ */
+class AggregateReportsWithHistoryTask extends AggregateReportsTask {
```

Comment:
I'm not sure if this new type of task is actually necessary. Why not to just extend the existing AggregateReportsTask and add there a condition that if some configuration option (e.g. "report { historical = true, html = true }" is enabled, then create an HTML report with historical data as well as create a snapshot. Clover's Ant <clover-html-reprot> task works in a similar way. https://confluence.atlassian.com/display/CLOVER/clover-html-report